bin/mlton-script

```diff
@@ -65,67 +49,100 @@ doitMLKit () {
 llvmOptOpt="-mem2reg -O2"
 # "New" Pass Manager; LLVM >= 13.0
 llvmOptOpt="--passes=function(mem2reg),default<O2>"
 
-doit "$lib" \
-        -ar-script "$lib/static-library"                         \
-        -cc "$CC"                                                \
-        -cc-opt '-std=gnu11 -fno-common'                         \
-        -cc-opt '-O1 -fno-strict-aliasing'                       \
-        -cc-opt '-foptimize-sibling-calls'                       \
-        -cc-opt '-w'                                             \
-        -cc-opt-quote "-I$lib/include"                           \
-        $gmpCCOpts $gmpLinkOpts                                  \
-        -llvm-llc-opt '-O2'                                      \
-        -llvm-opt-opt "$llvmOptOpt"                              \
-        -mlb-path-var 'SML_LIB $(LIB_MLTON_DIR)/sml'             \
-        -target-as-opt amd64 '-m64'                              \
-        -target-as-opt x86 '-m32'                                \
-        -target-cc-opt aix '-maix64'                             \
-        -target-cc-opt alpha                                     \
-                '-mieee -mbwx -mtune=ev6 -mfp-rounding-mode=d'   \
-        -target-cc-opt amd64 '-m64'                              \
-        -target-cc-opt amd64-darwin '-arch x86_64'               \
-        -target-cc-opt arm64-darwin '-arch arm64'                \
-        -target-cc-opt powerpc-darwin '-arch ppc'                \
-        -target-cc-opt powerpc64-darwin '-arch ppc64'            \
-        -target-cc-opt ia64-hpux "-mlp64"                        \
-        -target-cc-opt ia64 "-mtune=itanium2"                    \
-        -target-cc-opt sparc '-m32 -mcpu=v8 -Wa,-xarch=v8plusa'  \
-        -target-cc-opt x86 '-m32'                                \
-        -target-link-opt aix '-maix64'                           \
-        -target-link-opt alpha                                   \
-                '-mieee -mbwx -mtune=ev6 -mfp-rounding-mode=d'   \
-        -target-link-opt amd64 '-m64'                            \
-        -target-link-opt amd64-darwin '-arch x86_64'             \
-        -target-link-opt arm64-darwin '-arch arm64'              \
-        -target-link-opt powerpc-darwin '-arch ppc'              \
-        -target-link-opt powerpc64-darwin '-arch ppc64'          \
-        -target-link-opt ia64-hpux "-mlp64"                      \
-        -target-link-opt linux '-Wl,-znoexecstack'               \
-        -target-link-opt mingw                                   \
-                '-lws2_32 -lkernel32 -lpsapi -lnetapi32 -lwinmm' \
-        -target-link-opt mingw '-Wl,--enable-stdcall-fixup'      \
-        -target-link-opt solaris '-lnsl -lsocket -lrt'           \
-        -target-link-opt x86 '-m32'                              \
-        -profile-exclude '\$\(SML_LIB\)'                         \
-        "$@"
+# insert compile arguments
+looking='yes'
+for x; do
```

Comment:
`for arg; do` would be clearer.  `for arg in "$@"; do ` would be even clearer; I didn't realize that a `for` loop defaulted to iterating over the `$@`.